king_phisher/client/widget/managers.py

```diff
@@ -373,3 +374,61 @@ def signal_activate_popup_menu_copy(self, menuitem, column_ids):
 
 	def signal_activate_popup_menu_delete(self, menuitem):
 		self._call_cb_delete()
+
+class TimeSelector(gui_utilities.GladeGObject):
+	dependencies = gui_utilities.GladeDependencies(
+		children=(
+			'spinbutton_hour',
+			'spinbutton_minute'
+		),
+		top_level=(
+			'ClockHourAdjustment',
+			'ClockMinuteAdjustment'
+		)
+	)
+	top_gobject = 'popover'
+
+class TimeSelectorButtonManager(object):
+	def __init__(self, button, application, value=None):
+		"""
+		:param button: The button to activate TimeSelectorPopover()
+		:type button: :py:class:`Gtk.ToggleButton`
+		:param application: The application instance which this object belongs to.
+		:param value: The present datetime value (initialized to 00:00)
+		:type datetime.time: The value, (initialized to NoneType) will always return datetime.time(hr, min)
+		"""
+		self.popover = TimeSelector(application)
+		self.button = button
+		self.application = application
+		self._hour_spin = self.popover.gobjects['spinbutton_hour']
+		self._minute_spin = self.popover.gobjects['spinbutton_minute']
+		self.time = value or datetime.time(0, 0)
```

Comment:
to be on the safe side, I would recommed checking that the value is a datetime.time object prior to setting it to self.time.

Reply:
This is accomplished by the `value or datetime.time(0, 0)` because value will either be a NoneType or a datetime when it is read. If it evaluates a NoneType, it automatically reassigns it to `datetime.time(0, 0)` so the check is not necessary.
Thanks to @zeroSteiner for that little bit of py-fu.